tool/prelude.lua

```diff
@@ -42,7 +42,10 @@ function table.allvaluesbetweenincluding(seq, lowerbound, upperbound)
 end
 
 function table.containsduplicate(seq)
-  local seq2 = seq
+  local seq2 = {}
+  for i in pairs(seq) do
+      table.insert(seq2, i)
+  end
```

Comment:
```lua
> for i in pairs({5,9,11}) do print(i) end
1
2
3
```
```lua
for k,v in pairs({5,9,11}) do print(k,v) end
1	5
2	9
3	11
```
You want `for k,v in ipairs(seq) do table.insert(seq2, v) end` or according to [Lua Table Copy](http://lua-users.org/wiki/CopyTable) you can also use `{table.unpack(org)}`.

Reply:
I used the table.unpack(seq) suggestion as it takes less iterations.